### PR TITLE
Add PriorityClassName and Affinity

### LINKED ISSUE #284: Add PriorityClassName and Affinity to Helm Deployment Template
**What should the feature do:**
- Add PriorityClassName and Affinity to Helm Deployment Template

**What is use case behind this feature:**
- _PriorityClassName_: Enable specifying priorityClassName on controller pods (eg. system-cluster-critical) as user might need to [guarantee scheduling](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/) for controller pod within cluster
- _Affinity_:  Enable addition of [nodeAffinity, podAffinity, podAntiAffinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node). This will further improve resiliency of the pvc-autoresizer deployment together with the implemented leader election and  available configuration for replicas.

### PR DESCRIPTION
**Pull request by josephangbc**
closes #284

### CODE REVIEW

File: charts/pvc-autoresizer/values.yaml [hunk 1]
```diff
@@ -72,6 +72,33 @@ controller:
...
+    #       podAffinityTerm:
+    #         labelSelector:
+    #           matchExpressions:
+    #           - key: app.kubernetes.io/name: pvc-autoresizer
+    #             operator: In
+    #             values:
+    #             - pvc-autoresizer
+    #         topologyKey: topology.kubernetes.io/zone
+
+  # controller.priorityClassName -- riority class name to be applied to the controller pods.
```
Comment (toshipp) [inline charts/pvc-autoresizer/values.yaml:98]:
```suggestion
  # controller.priorityClassName -- priority class name to be applied to the controller pods.
```

File: charts/pvc-autoresizer/values.yaml [hunk 2]
```diff
@@ -72,6 +72,33 @@ controller:
...
+    # podAffinity:
+    #     requiredDuringSchedulingIgnoredDuringExecution:
+    #     - labelSelector:
+    #         matchExpressions:
+    #         - key: app.kubernetes.io/name: pvc-autoresizer
+    #           operator: In
+    #           values:
+    #           - pvc-autoresizer
+    #       topologyKey: topology.kubernetes.io/zone
+    #   podAntiAffinity:
```
Comment (toshipp) [inline charts/pvc-autoresizer/values.yaml:86]: It seems to be the wrong indentation.

File: charts/pvc-autoresizer/templates/controller/deployment.yaml [hunk 3]
```diff
@@ -98,3 +98,10 @@ spec:
       {{- end }}
       securityContext:
         {{- toYaml .Values.controller.podSecurityContext | nindent 8 }}
+    {{- with .Values.controller.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+    {{- if .Values.controller.priorityClassName }}
```
Comment (toshipp) [inline charts/pvc-autoresizer/templates/controller/deployment.yaml:105]: Could you use `with` instead?

**Review by toshipp — CHANGES_REQUESTED** [review]
almost good. I left some comments, please check it.

File: charts/pvc-autoresizer/values.yaml (hunk 1 above)
Comment (josephangbc) [reply charts/pvc-autoresizer/values.yaml]: Updated the example comments

File: charts/pvc-autoresizer/values.yaml (hunk 2 above)
Comment (josephangbc) [reply charts/pvc-autoresizer/values.yaml]: Updated the indentation

**Review by satoru-takeuchi — APPROVED** [review]

**Comment by toshipp** [comment]
@josephangbc 
Could you re-generate `charts/pvc-autoresizer/README.md`?
The instruction is `make generate-helm-docs`. If you didn't setup tools, please `make setup` first.

**Review by toshipp — APPROVED** [review]

**Review by satoru-takeuchi — APPROVED** [review]

**Pull request merged**